informer/informer.go

```diff
@@ -339,6 +350,52 @@ func (i *Informer) isCachedFilled() bool {
 	return false
 }
 
+func (i *Informer) newWatcher(ctx context.Context) (watch.Interface, error) {
+	var err error
+
+	var watcher watch.Interface
+	{
+		o := func() error {
+			failed := make(chan error, 1)
+			found := make(chan struct{}, 1)
+
+			go func() {
+				watcher, err = i.watcher.Watch(i.listOptions)
+				if err != nil {
+					failed <- microerror.Mask(err)
+					return
+				}
+
+				found <- struct{}{}
+			}()
+
+			select {
+			case <-ctx.Done():
+				return backoff.Permanent(microerror.Mask(contextCanceledError))
+			case err := <-failed:
+				return microerror.Mask(err)
+			case <-found:
+				// fall through
+			case <-time.After(time.Second):
+				return microerror.Mask(initializationTimedOutError)
+			}
```

Comment:
Shouldn't we close something in case or errors or timeouts?

Reply:
The watcher is nil in all cases but `found`. So closing it would cause a panic AFAIK. I just thought that the channels are not collected and thus the goroutine of the watcher constructor leaks in case the channels are not closed.

Reply:
Ack. No channels are GC-ed like any other variable.